packages/grafana-ui/src/components/Input/Input.tsx

```diff
@@ -171,7 +171,8 @@ export const getInputStyles = stylesFactory(({ theme, invalid = false, width }:
         &:not(:first-child) {
           padding-left: ${prefixSuffixStaticWidth};
         }
-        &:not(:last-child) {
+        &:not(:last-child),
+        &:not(& + div > [data-testid='Spinner']) {
```

Comment:
any other way to solve this? it's pretty unexpected to use a data-test id for styling :) if you really need this use a manually specified classname (like `autosave-loading-spinner`)

Reply:
> any other way to solve this? it's pretty unexpected to use a data-test id for styling :) if you really need this use a manually specified classname (like `autosave-loading-spinner`)

Thanks @torkelo for the idea! I was working on this right now as I didn't like this solution either.